tests/ClientTest.php

```diff
@@ -336,6 +338,68 @@ public function sampleRateAbsoluteDataProvider(): array
         ];
     }
 
+    public function testProcessDiscardsEventWhenItIsSampledDueToSampleRateOption(): void
+    {
+        /** @var LoggerInterface&MockObject $logger */
```

Comment:
Do we really need those PHPDoc annotations?

Reply:
Yes, as said multiple times in the past they help the IDE in autocompleting things and stopping warnings about types mismatch. Sure, there is a plugin for PHPStorm that does it without annotations, but it has not been updated since ages so it may cease to work at some point. Also this is not a PHPStorm-specific annotation, so it may help people using other IDEs too